File: peek_plugin_eventdb/_private/tuples/AdminStatusTuple.py
from vortex.Tuple import addTupleType, TupleField, Tuple

from peek_plugin_eventdb._private.PluginNames import eventdbTuplePrefix


@addTupleType
class AdminStatusTuple(Tuple):
    __tupleType__ = eventdbTuplePrefix + "AdminStatusTuple"

    rawValueQueueStatus: bool = TupleField(False)
    rawValueQueueSize: int = TupleField(0)
    rawValueProcessedTotal: int = TupleField(0)
    rawValueLastError: str = TupleField()

File: peek_plugin_eventdb/_private/admin-app/tuples/AdminStatusTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {eventdbTuplePrefix} from "@peek/peek_plugin_eventdb/_private/PluginNames";


@addTupleType
export class AdminStatusTuple extends Tuple {
    public static readonly tupleName = eventdbTuplePrefix + "AdminStatusTuple";

    rawValueQueueStatus: boolean;
    rawValueQueueSize: number;
    rawValueProcessedTotal: number;
    rawValueLastError: string;

    constructor() {
        super(AdminStatusTuple.tupleName)
    }
}